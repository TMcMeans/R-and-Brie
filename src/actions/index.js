export const addRecipes = recipes => ({
  type: 'ADD_RECIPES',
  recipes: recipes
});

export const hasErrored = message => ({
  type: 'HAS_ERRORED',
  message
});

export const isLoading = bool => ({
  type: 'IS_LOADING',
  isLoading: bool
});

export const addReview = review => ({
  type: 'ADD_REVIEW',
  review: review
})