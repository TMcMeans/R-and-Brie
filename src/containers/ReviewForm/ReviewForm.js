import React, { Component } from 'react';
import StarRatingComponent from 'react-star-rating-component';

class ReviewForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      review: '',
      rating: 0
    };
  }

  onStarClick = (nextValue, prevValue, name) => {
    this.setState({ rating: nextValue });
  };

  render() {
    const { rating } = this.state;

    return (
      <div className="review">
        <form action="">
          <input type="text" placeholder="Add a comment..." />
          <h2>Rating from state: {rating}</h2>
          <StarRatingComponent
            name="rate1"
            starCount={5}
            value={rating}
            onStarClick={this.onStarClick}
            starColor={`#ffb400`}
            emptyStarColor={`#000`}
          />
          <button>Post</button>
        </form>
      </div>
    );
  }
}

export default ReviewForm;
