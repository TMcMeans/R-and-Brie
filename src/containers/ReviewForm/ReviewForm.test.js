import React from 'react';
import { shallow } from 'enzyme'

import { ReviewForm } from '../ReviewForm/ReviewForm'
import { mapStateToProps, mapDispatchToProps } from '../ReviewForm/ReviewForm'
import { addReview } from '../../actions'

describe('ReviewForm', () => {
  let mockReviews = [{ label: 'Baked brie', rating: 4, caption: 'Tastes great!' }]
  let wrapper
  it('should match snapshot with correct data', () => {
    wrapper = shallow(<ReviewForm reviews={[{ caption: 'loved it', rating: 4, label: 'brie croissants', name: 'Tanjie' }]} addReview={jest.fn()} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should setState when onStarClick is called', () => {
    wrapper = shallow(<ReviewForm reviews={[]} addReview={jest.fn()} />)
    wrapper.instance().onStarClick(1)
    expect(wrapper.state('rating')).toEqual(1)
  })
  it.skip('should call setState when handleChange is called', () => {
    wrapper = shallow(<ReviewForm reviews={[]} addReview={jest.fn()} />)
    const mockEvent = { target: { value: 'Tanjie' } }

    wrapper.instance().handleChange(mockEvent)

    expect(wrapper.state('name')).toEqual('Tanjie')
    expect(wrapper.state('caption')).toEqual('I loved it')

  })

  describe('mapStateToProps', () => {
    it('should return an array of reviews', () => {
      const mockReviews = [{ caption: 'I loved it', label: 'baked feta' }, { caption: 'I hated it', label: 'brie croissants' }]

      const mockState = {
        reviews: mockReviews
      }

      const expected = { reviews: mockReviews }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)

    })
  })

  describe('mapDispatchToProps', () => {
    it('should dispatch addReview when handleSubmit is called', () => {
      const mockDispatch = jest.fn()
      const mockReview = { caption: 'Hated it', label: 'brie croissants' }

      const actionToDispatch = addReview(mockReview)

      const mappedProps = mapDispatchToProps(mockDispatch)

      mappedProps.addReview({ caption: 'Hated it', label: 'brie croissants' })
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})


