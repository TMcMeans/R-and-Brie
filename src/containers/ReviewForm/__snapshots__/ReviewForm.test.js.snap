// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewForm should match snapshot with correct data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReviewForm
    addReview={[MockFunction]}
    reviews={
      Array [
        Object {
          "caption": "loved it",
          "label": "brie croissants",
          "name": "Tanjie",
          "rating": 4,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
      >
        <input
          name="name"
          onChange={[Function]}
          placeholder="Enter your name"
          type="text"
          value=""
        />
        <input
          name="caption"
          onChange={[Function]}
          placeholder="Add a comment..."
          type="text"
          value=""
        />
        <h2>
          Rating from state: 
          0
        </h2>
        <StarRatingComponent
          className="star-ratings"
          editing={true}
          emptyStarColor="#000"
          name="rate1"
          onStarClick={[Function]}
          starColor="#ffb400"
          starCount={5}
          value={0}
        />
        <input
          type="submit"
        />
      </form>,
      "className": "review",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="name"
            onChange={[Function]}
            placeholder="Enter your name"
            type="text"
            value=""
          />,
          <input
            name="caption"
            onChange={[Function]}
            placeholder="Add a comment..."
            type="text"
            value=""
          />,
          <h2>
            Rating from state: 
            0
          </h2>,
          <StarRatingComponent
            className="star-ratings"
            editing={true}
            emptyStarColor="#000"
            name="rate1"
            onStarClick={[Function]}
            starColor="#ffb400"
            starCount={5}
            value={0}
          />,
          <input
            type="submit"
          />,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "name",
            "onChange": [Function],
            "placeholder": "Enter your name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "caption",
            "onChange": [Function],
            "placeholder": "Add a comment...",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Rating from state: ",
              0,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Rating from state: ",
            0,
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "star-ratings",
            "editing": true,
            "emptyStarColor": "#000",
            "name": "rate1",
            "onStarClick": [Function],
            "starColor": "#ffb400",
            "starCount": 5,
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "type": "submit",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
        >
          <input
            name="name"
            onChange={[Function]}
            placeholder="Enter your name"
            type="text"
            value=""
          />
          <input
            name="caption"
            onChange={[Function]}
            placeholder="Add a comment..."
            type="text"
            value=""
          />
          <h2>
            Rating from state: 
            0
          </h2>
          <StarRatingComponent
            className="star-ratings"
            editing={true}
            emptyStarColor="#000"
            name="rate1"
            onStarClick={[Function]}
            starColor="#ffb400"
            starCount={5}
            value={0}
          />
          <input
            type="submit"
          />
        </form>,
        "className": "review",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="name"
              onChange={[Function]}
              placeholder="Enter your name"
              type="text"
              value=""
            />,
            <input
              name="caption"
              onChange={[Function]}
              placeholder="Add a comment..."
              type="text"
              value=""
            />,
            <h2>
              Rating from state: 
              0
            </h2>,
            <StarRatingComponent
              className="star-ratings"
              editing={true}
              emptyStarColor="#000"
              name="rate1"
              onStarClick={[Function]}
              starColor="#ffb400"
              starCount={5}
              value={0}
            />,
            <input
              type="submit"
            />,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "name",
              "onChange": [Function],
              "placeholder": "Enter your name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "caption",
              "onChange": [Function],
              "placeholder": "Add a comment...",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Rating from state: ",
                0,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Rating from state: ",
              0,
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "star-ratings",
              "editing": true,
              "emptyStarColor": "#000",
              "name": "rate1",
              "onStarClick": [Function],
              "starColor": "#ffb400",
              "starCount": 5,
              "value": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "type": "submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
